Debugger does not work on Elixir 1.10

### Environment

* Elixir & Erlang versions (elixir --version):
>Erlang/OTP 22 [erts-10.6.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
>
>Elixir 1.10.2 (compiled with Erlang/OTP 22)
* Operating system:
>System Version:	macOS 10.15.3 (19D76)
>Kernel Version:	Darwin 19.3.0
* Editor or IDE name (e.g. Emacs/VSCode): VSCode
* LSP Client name: 
* If using VSCode, are you using "ElixirLS Fork: Elixir support and debugger" (note that it has "Fork" in the name)? Yes

### Troubleshooting

- [x] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [x] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor
  * NOTE: This will cause you to have to re-run the entire dialyzer build

### Report

No matter what I do, I can't seem to get the debugger to act the same as when I run `mix phx.server` on the command line.

Here's my `launch.json` file:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "mix_task",
            "name": "mix phx.server",
            "task": "phx.server",
            "request": "launch",
            "projectDir": "${workspaceRoot}",
            "startApps": true
        }
    ]
}
```
I have set `startApps` to `true` because if I don't then I get errors about opencensus.

My `config/dev.exs` has the following at the beginning, so the server starts on port 4000 and it kicks off `npm start` which runs my webpack-dev-server setup.
```
config :app_name, AppNameWeb.Endpoint,
  http: [port: 4000],
  debug_errors: true,
  code_reloader: true,
  check_origin: false,
  watchers: [
    npm: ["start"]
  ]

config :app_name, AppNameWeb.Endpoint,
  live_reload: [
    patterns: [
      # ~r{priv/static/.*(js|css|png|jpeg|jpg|gif|svg)$},
      ~r{lib/app_name_web/views/.*(ex)$},
      ~r{lib/app_name_web/templates/.*(eex)$},
      ~r{lib/app_name_web/live/.*(ex)$}
    ]
  ]
```

However, when I run the debugger, NPM doesn't seem to get started and no server ever seems to come up on port 4000.  I know that `config/dev.exs` is being run because when I add a `IO.inspect` line at the top, I see my message printed in the Debug Console in VSCode.  If I place a `IO.inspect` line beneath it, that gets printed as well, so I know nothing is erroring out from that config call.

I also always get errors related to my [Oban](https://github.com/sorentwo/oban) and [TelemetryMetricsPrometheus](https://github.com/beam-telemetry/telemetry_metrics_prometheus) setup.

```2020-03-20T22:29:35.304Z [ERROR] GenServer Oban.Queue.Conversion.Producer terminating
** (UndefinedFunctionError) function DBConnection.past_event/3 is undefined or private
    (db_connection 2.2.1) DBConnection.past_event(nil, :checkin, -576460721230932000)
    (db_connection 2.2.1) lib/db_connection.ex:1037: DBConnection.checkout/3
    (db_connection 2.2.1) lib/db_connection.ex:711: DBConnection.run/3
    (stdlib 3.11.2) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.11.2) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: :poll
State: %Oban.Queue.Producer.State{circuit: :enabled, conf: %Oban.Config{beats_maxage: 300, circuit_backoff: 30000, crontab: [{%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [:*], months: [:*], weekdays: [:*]}, AppName.ThirdParty.Sync, []}, {%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55], months: [:*], weekdays: [:*]}, AppName.DataProcessing.Runner, [args: %{check: true}]}], dispatch_cooldown: 5, name: Oban, node: "FR-A013", poll_interval: 1000, prefix: "public", prune: {:maxage, 259200}, prune_interval: 60000, prune_limit: 5000, queues: [default: 10, conversion: 10, processing: 10, third_party_sync: 10], repo: AppName.Repo, rescue_after: 60, rescue_interval: 60000, shutdown_grace_period: 15000, timezone: "Etc/UTC", verbose: false}, cooldown_ref: nil, dispatched_at: -576460720228, foreman: Oban.Queue.Conversion.Foreman, limit: 10, name: Oban.Queue.Conversion.Producer, nonce: "uhdfl0ju", paused: false, poll_ref: #Reference<0.1730123617.4046716929.79534>, queue: "conversion", rescue_ref: #Reference<0.1730123617.4046716929.44865>, running: %{}, started_at: ~U[2020-03-20 22:29:17.081743Z]}
2020-03-20T22:29:35.304Z [ERROR] GenServer Oban.Queue.Default.Producer terminating
** (UndefinedFunctionError) function DBConnection.past_event/3 is undefined or private
    (db_connection 2.2.1) DBConnection.past_event(nil, :checkin, -576460721231309000)
    (db_connection 2.2.1) lib/db_connection.ex:1037: DBConnection.checkout/3
    (db_connection 2.2.1) lib/db_connection.ex:711: DBConnection.run/3
    (stdlib 3.11.2) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.11.2) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: :poll
State: %Oban.Queue.Producer.State{circuit: :enabled, conf: %Oban.Config{beats_maxage: 300, circuit_backoff: 30000, crontab: [{%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [:*], months: [:*], weekdays: [:*]}, AppName.ThirdParty.Sync, []}, {%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55], months: [:*], weekdays: [:*]}, AppName.DataProcessing.Runner, [args: %{check: true}]}], dispatch_cooldown: 5, name: Oban, node: "FR-A013", poll_interval: 1000, prefix: "public", prune: {:maxage, 259200}, prune_interval: 60000, prune_limit: 5000, queues: [default: 10, conversion: 10, processing: 10, third_party_sync: 10], repo: AppName.Repo, rescue_after: 60, rescue_interval: 60000, shutdown_grace_period: 15000, timezone: "Etc/UTC", verbose: false}, cooldown_ref: nil, dispatched_at: -576460720228, foreman: Oban.Queue.Default.Foreman, limit: 10, name: Oban.Queue.Default.Producer, nonce: "e02gifo1", paused: false, poll_ref: #Reference<0.1730123617.4046716930.34527>, queue: "default", rescue_ref: #Reference<0.1730123617.4046716929.44850>, running: %{}, started_at: ~U[2020-03-20 22:29:17.077948Z]}
2020-03-20T22:29:35.304Z [ERROR] GenServer Oban.Queue.ThirdPartySync.Producer terminating
** (UndefinedFunctionError) function DBConnection.past_event/3 is undefined or private
    (db_connection 2.2.1) DBConnection.past_event(nil, :checkin, -576460722273411000)
    (db_connection 2.2.1) lib/db_connection.ex:1037: DBConnection.checkout/3
    (db_connection 2.2.1) lib/db_connection.ex:711: DBConnection.run/3
    (stdlib 3.11.2) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.11.2) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: :poll
State: %Oban.Queue.Producer.State{circuit: :enabled, conf: %Oban.Config{beats_maxage: 300, circuit_backoff: 30000, crontab: [{%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [:*], months: [:*], weekdays: [:*]}, AppName.ThirdParty.Sync, []}, {%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55], months: [:*], weekdays: [:*]}, AppName.DataProcessing.Runner, [args: %{check: true}]}], dispatch_cooldown: 5, name: Oban, node: "FR-A013", poll_interval: 1000, prefix: "public", prune: {:maxage, 259200}, prune_interval: 60000, prune_limit: 5000, queues: [default: 10, conversion: 10, processing: 10, third_party_sync: 10], repo: AppName.Repo, rescue_after: 60, rescue_interval: 60000, shutdown_grace_period: 15000, timezone: "Etc/UTC", verbose: false}, cooldown_ref: nil, dispatched_at: -576460720228, foreman: Oban.Queue.ThirdPartySync.Foreman, limit: 10, name: Oban.Queue.ThirdPartySync.Producer, nonce: "2m4fcr3p", paused: false, poll_ref: #Reference<0.1730123617.4046716931.35804>, queue: "third_party_sync", rescue_ref: #Reference<0.1730123617.4046716929.44856>, running: %{}, started_at: ~U[2020-03-20 22:29:17.082110Z]}
2020-03-20T22:29:35.304Z [ERROR] GenServer Oban.Queue.Processing.Producer terminating
** (UndefinedFunctionError) function DBConnection.past_event/3 is undefined or private
    (db_connection 2.2.1) DBConnection.past_event(nil, :checkin, -576460722273383000)
    (db_connection 2.2.1) lib/db_connection.ex:1037: DBConnection.checkout/3
    (db_connection 2.2.1) lib/db_connection.ex:711: DBConnection.run/3
    (stdlib 3.11.2) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.11.2) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: :poll
State: %Oban.Queue.Producer.State{circuit: :enabled, conf: %Oban.Config{beats_maxage: 300, circuit_backoff: 30000, crontab: [{%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [:*], months: [:*], weekdays: [:*]}, AppName.ThirdParty.Sync, []}, {%Oban.Crontab.Cron{days: [:*], hours: [:*], minutes: [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55], months: [:*], weekdays: [:*]}, AppName.DataProcessing.Runner, [args: %{check: true}]}], dispatch_cooldown: 5, name: Oban, node: "FR-A013", poll_interval: 1000, prefix: "public", prune: {:maxage, 259200}, prune_interval: 60000, prune_limit: 5000, queues: [default: 10, conversion: 10, processing: 10, third_party_sync: 10], repo: AppName.Repo, rescue_after: 60, rescue_interval: 60000, shutdown_grace_period: 15000, timezone: "Etc/UTC", verbose: false}, cooldown_ref: nil, dispatched_at: -576460720228, foreman: Oban.Queue.Processing.Foreman, limit: 10, name: Oban.Queue.Processing.Producer, nonce: "355krvf4", paused: false, poll_ref: #Reference<0.1730123617.4046716931.35805>, queue: "processing", rescue_ref: #Reference<0.1730123617.4046716930.27017>, running: %{}, started_at: ~U[2020-03-20 22:29:17.081962Z]}
2020-03-20T22:29:35.322Z [INFO] Postgrex.Protocol (#PID<0.2952.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3132.0> exited
2020-03-20T22:29:35.323Z [INFO] Postgrex.Protocol (#PID<0.2955.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3128.0> exited
2020-03-20T22:29:35.323Z [INFO] Postgrex.Protocol (#PID<0.2951.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3124.0> exited
2020-03-20T22:29:35.324Z [INFO] Postgrex.Protocol (#PID<0.2953.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.3120.0> exited
2020-03-20T22:29:51.004Z [ERROR] Handler {TelemetryMetricsPrometheus.Core.EventHandler, #PID<0.3005.0>, [:vm, :memory, :total]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:29:51.004Z [ERROR] Handler {TelemetryMetricsPrometheus.Core.EventHandler, #PID<0.3005.0>, [:vm, :total_run_queue_lengths, :total]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:29:51.004Z [ERROR] Handler {TelemetryMetricsPrometheus.Core.EventHandler, #PID<0.3005.0>, [:vm, :total_run_queue_lengths, :cpu]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:29:51.005Z [ERROR] Handler {TelemetryMetricsPrometheus.Core.EventHandler, #PID<0.3005.0>, [:vm, :total_run_queue_lengths, :io]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:30:17.160Z [ERROR] Handler "oban-default-logger" has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:30:17.160Z [ERROR] Handler "oban-default-logger" has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

2020-03-20T22:30:17.196Z [ERROR] Handler {TelemetryMetricsPrometheus.Core.EventHandler, #PID<0.3005.0>, [:oban, :success, :duration]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]
```

I've tried killing the code that sets up the telemetry library and while it stops these errors from happening, it doesn't fix the underlying problem where the server never starts.  Also, there's clearly something happening since the debugging sessions doesn't end with an error or anything like that.  And I see various processes/tasks/agents/genservers/whatever running in the callstack pane in VSCode.  I'm not sure what to do to set this up properly.

FWIW, I'm able to set up a new boilerplate project using `phx_new` that will cooperate with the debugger relatively fine, although I do get these errors in the console:
```
[error] Handler {Phoenix.Logger, [:phoenix, :endpoint, :start]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

[error] Handler {Phoenix.Logger, [:phoenix, :router_dispatch, :start]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]

[error] Handler {Phoenix.Logger, [:phoenix, :endpoint, :stop]} has failed and has been detached. Class=:error
Reason=:undef
Stacktrace=[]
```

--- a/apps/debugger/lib/debugger/cli.ex
+++ b/apps/debugger/lib/debugger/cli.ex
@@ -8,6 +8,22 @@ defmodule ElixirLS.Debugger.CLI do
     Application.ensure_all_started(:debugger, :permanent)
     IO.puts("Started ElixirLS debugger v#{Launch.debugger_version()}")
     Launch.print_versions()
+    warn_if_unsupported_version()
     WireProtocol.stream_packets(&Server.receive_packet/1)
+  end
+
+  # Debugging does not work on Elixir 1.10.0-1.10.2:
+  # https://github.com/elixir-lsp/elixir-ls/issues/158
+  defp warn_if_unsupported_version do
+    elixir_version = System.version()
+
+    if Version.match?(elixir_version, ">= 1.10.0") && Version.match?(elixir_version, "< 1.10.3") do
+      message =
+        "WARNING: Debugging is not supported on Elixir #{elixir_version}. Please upgrade" <>
+          " to at least 1.10.3\n" <>
+          "more info: https://github.com/elixir-lsp/elixir-ls/issues/158"
+
+      Output.print_err(message)
+    end
   end
 end

